client.py

```diff
@@ -1 +1,47 @@
-# I'm empty!
+#!/usr/bin/python
+import sys, re, http.client, urllib.parse
+
+def main():
+    args = sys.argv[1:]
+    if len(args) != 4:
+        print ("ERROR: Invalid number of arguments")
+        return 1
+    
+    ip = re.match(IP_PATTERN, args[0], flags=0)
```

Comment:
Regular expressions are expensive to keep recompiling. Because the number of regexs is small, however, I believe the compilations will be cached in this instance. It may be smart to pre-compile them just to be safe though. It'd look like this:

`ip = re.compile("\d{3}\.\d{3}\.\d(\d{1,2})?\.\d(\d{1,2})?", flags=0)
ip.match(args[0])`

Reply:
After thinking through it, this change would do absolutely nothing for our program performance, since the client is being executed anew with each request.